docs/dev/table/connectors/index.md

```diff
@@ -0,0 +1,268 @@
+      <td>Streaming Sink, Batch Sink</td>
+    </tr>
+    </tbody>
+</table>
+
+{% top %}
+
+How to use connectors
+--------
+
+Flink supports to use SQL CREATE TABLE statement to register a table. One can define the name of the table, the schema of the table, the connector options for connecting to an external system.
+
+The following code shows a full example of how to connect to Kafka for reading Json records.
+
+<div class="codetabs" markdown="1">
+<div data-lang="SQL" markdown="1">
+{% highlight sql %}
+CREATE TABLE MyUserTable (
+  -- declare the schema of the table
+  `user` BIGINT,
+  message STRING,
+  ts TIMESTAMP,
+  proctime AS PROCTIME(), -- use computed column to define proctime attribute
+  WATERMARK FOR ts AS ts - INTERVAL '5' SECOND  -- use WATERMARK statement to define rowtime attribute
+) WITH (
+  -- declare the external system to connect to
+  'connector' = 'kafka',
+  'topic' = 'topic_name',
+  'scan.startup.mode' = 'earliest-offset',
+  'properties.bootstrap.servers' = 'localhost:9092',
+  'format' = 'json'   -- declare a format for this system
+)
+{% endhighlight %}
+</div>
+</div>
+
+In this ways the desired connection properties are converted into normalized, string-based key-value pairs. So-called [table factories](sourceSinks.html#define-a-tablefactory) create configured table sources, table sinks, and corresponding formats from the key-value pairs. All table factories that can be found via Java's [Service Provider Interfaces (SPI)](https://docs.oracle.com/javase/tutorial/sound/SPI-intro.html) are taken into account when searching for exactly-one matching table factory.
+
+If no factory can be found or multiple factories match for the given properties, an exception will be thrown with additional information about considered factories and supported properties.
+
+{% top %}
+
+Schema Mapping
+------------
+
+The body clause of a SQL `CREATE TABLE` statement defines the names and types of columns, and constraints, watermarks. Flink doesn't hold the data, thus the schema definition only declares how to map types from an external system to Flink’s representation. The mapping may not be mapped by names, it depends on the implementation of formats and connectors. For example, a MySQL database table is mapped by field names (not case sensitive), and a CSV filesystem is mapped by field order (field names can be arbitrary). This will be explanation in every connectors.
+
+The following example shows a simple schema without time attributes and one-to-one field mapping of input/output to table columns.
+
+<div class="codetabs" markdown="1">
+<div data-lang="SQL" markdown="1">
+{% highlight sql %}
+CREATE TABLE MyTable (
+  MyField1 INT,
+  MyField2 STRING,
+  MyField3 BOOLEAN
+) WITH (
+  ...
+)
+{% endhighlight %}
+</div>
+</div>
+
+### Primary Key
+
+Primary key constraints tell that a column or a set of columns of a table are unique and they do not contain null. Primary key therefore uniquely identify a row in a table.
+
```

Comment:
`contain null` -> `contain nulls`. Remove the therefore : `Primary key uniquely identifies`